dbt/adapters/databricks/impl.py

```diff
@@ -376,6 +384,36 @@ def parse_describe_extended(  # type: ignore[override]
 
     def get_columns_in_relation(  # type: ignore[override]
         self, relation: DatabricksRelation
+    ) -> List[DatabricksColumn]:
+        if (
+            # We can uncomment this once behavior flags are available to adapters
+            # self.behavior.column_types_from_information_schema and # type: ignore
+            not relation.is_hive_metastore()
```

Comment:
Is there a reason we only enable it for UC now? Should we just enable for non-UC to avoid bug?

Reply:
We only have information_schema for UC last I checked.